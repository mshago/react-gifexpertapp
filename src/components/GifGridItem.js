import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({ id, title, url, time=0 }) => {

    return (
        <div className={'card animate__animated animate__fadeIn animate__delay-'+time+'s'}>
           <img src={ url } alt={ title } />
           <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

export default GifGridItem;