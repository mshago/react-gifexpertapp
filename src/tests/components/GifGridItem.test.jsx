import React from 'react'
import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('Tests on <GifGridItem />',() => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.png';
    const wrapper = shallow(
        <GifGridItem 
            title={title}
            url={url}
        /> 
    );

    test('shows the component correctly',() => {
        expect(wrapper).toMatchSnapshot();
    });

    test('must have a paragraph with a title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    });

    test('match img', () => {
        const img = wrapper.find('img');
        expect( img.prop('src')).toBe(url);
        expect( img.prop('alt') ).toBe(title);
    });

    test('must have animate__fadeIn class',() => {
        const div = wrapper.find('div')
        expect(div.hasClass('animate__fadeIn'));
    })
});