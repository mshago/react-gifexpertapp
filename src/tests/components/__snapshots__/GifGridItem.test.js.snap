// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGridItem /> shows the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-0s"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.png"
  />
  <p>
    Un titulo
  </p>
</div>
`;
